src/main/java/com/fasterxml/jackson/databind/introspect/AnnotatedCreatorCollector.java

```diff
@@ -241,14 +248,99 @@ Creators collect(Class<?> primaryMixIn)
         for (int i = 0; i < factoryCount; ++i) {
             AnnotatedMethod factory = result.get(i);
             if (factory == null) {
+                Method candidate = candidates.get(i);
+                TypeResolutionContext contextForFactory = emptyTypeResCtxt;
+                propagateTypeParameters:
```

Comment:
TODO: replace named block+break with a function with a nullable result for readability.